pulsar-broker/src/test/java/org/apache/pulsar/broker/service/BrokerServiceTest.java

```diff
@@ -1056,7 +1105,12 @@ public void testLookupThrottlingForClientByClient() throws Exception {
             long reqId10 = reqId++;
             ByteBuf request10 = Commands.newLookup(topicName, true, reqId10);
             f3 = pool.getConnection(resolver.resolveHost())
-                .thenCompose(clientCnx -> clientCnx.newLookup(request10, reqId10));
+                .thenCompose(clientCnx -> {
+                    CompletableFuture<?> future = clientCnx.newLookup(request10, reqId10);
+                    // pending other responses in `ClientCnx` until now
+                    latch[0].countDown();
```

Comment:
Why not code like this (Simpler and better understood):

```java
latch[0] = new CountDownLatch(1);

f1 = pool.getConnection(resolver.resolveHost()).thenCompose(clientCnx -> clientCnx.newLookup(request8,reqId8));
f2 = pool.getConnection(resolver.resolveHost()).thenCompose(clientCnx -> clientCnx.newLookup(request9, reqId9));
f3 = pool.getConnection(resolver.resolveHost()).thenCompose(clientCnx -> clientCnx.newLookup(request9, reqId9));

latch[0].countDown();

f1.get();
f2.get();
f3.get();
fail("At least one should fail");
```

Reply:
Hi @AnonHxy and this one comment :)

Reply:
I think `f1` and `f2` and `f3` could all complete successfully if we do so. The reason is that:

`clientCnx.newLookup` will be called in another thread, so `latch[0].countDown()` could happen before all `newLookup` requeset sending, that will cause the latch has no effect in this test.

In order to make sure `f3.get()` fail, we must block handing response in `clientCnx` until the third `newLookup` have send.  @poorbarcode

Reply:
I'm sorry I forgot the `thenCompose`